chore(NA): correctly set kibanaRoot on kibana plugin helpers for new generated plugins

mistic opened the pull request:
That PR is a small fix continuing the work done here https://github.com/elastic/kibana/pull/31748

Right now if you generate a plugin with our plugin generator script and try to start it, an error would be given as an output as the plugin helpers would calculate the kibana root directory as if the plugin has been generated into `kibana-extra` other than to the new location `kibana_root/plugins`.

That PR fix the problem and make it possible to use both directories. However it adds a warning message to the console so the users would know it is time to move on to the new preferred path

elasticmachine commented:
Pinging @elastic/kibana-operations

elasticmachine commented:
Pinging @elastic/kibana-platform

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10884/)

mistic commented:
retest

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10893/)

joshdover commented:
Will this break plugins that are still in kibana-extra?

mistic commented:
I think that would only break plugins that are in the `kibana-extra` plugin and will try to developing against those new versions where we would deploy this right? This would only affect the `start` command from `kbn-plugin-helpers`

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10927/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10956/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11431/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11918/)

joshdover commented:
> I think that would only break plugins that are in the kibana-extra plugin and will try to developing against those new versions where we would deploy this right?

Correct, but we decided to keep allowing kibana-extra to work for some time: https://github.com/elastic/kibana/pull/31748#issuecomment-469717015

I think we should either (1) push this to 7.4 instead of 7.3.1; or (2) make it work with both kibana-extra and plugins directories. Additionally, it looks like [this error message](https://github.com/elastic/kibana/pull/31748#issuecomment-506269844) should updated: `Error: The `kibanaRoot` config option has been removed from `@kbn/plugin-helpers`. During development your plugin must be located in `../kibana-extra/{pluginName}` relative to the Kibana directory to work with this package.
`

If we decide to push this to 7.4 rather than support both directories, we should include a dev docs note and a informative error message for plugins that are still using kibana-extra.

mistic commented:
@joshdover thanks for the comments. I've updated the PR and I just found an easy way to keep both. I've updated the error message and also added a warning message to let the users now that is time to move the plugins to the new preferred path. So nothing breaking behaviour would be introduced.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12074/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12078/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/12145/)

mistic commented:
7.x: pending
7.4: pending
